File: client-app/src/app/types/customer.ts
export interface Customer {
	id: string;
	name: string;
	email: string;
	tel: string;
	description: string;
	regDate: Date | null;
	courseId: string;
	courseName: string;
}

export class Customer implements Customer {
	id = '';
	name = '';
	email = '';
	tel = '';
	description = '';
	courseId = '';
	courseName = '';
	regDate: Date | null = new Date();
	constructor(customer?: Customer) {
		if (customer) {
			Object.assign(this, customer);
		}
	}
}

File: API/Entities/Customer.cs
namespace API.Entities;

public class Customer
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string Email { get; set; }
    public string Tel { get; set; }
    public Course Course { get; set; }
    public string Description { get; set; }
    public DateTime RegDate { get; set; }

}
